examples/gno.land/p/demo/json/decode.gno

```diff
@@ -0,0 +1,289 @@
+// ref: https://github.com/spyzhov/ajson/blob/master/decode.go
+
+package json
+
+import (
+	"errors"
+	"io"
+
+	"gno.land/p/demo/ufmt"
+)
+
+// This limits the max nesting depth to prevent stack overflow.
+// This is permitted by https://tools.ietf.org/html/rfc7159#section-9
+const maxNestingDepth = 10000
+
+// Unmarshal parses the JSON-encoded data and returns a Node.
+// The data must be a valid JSON-encoded value.
+//
+// Usage:
+//
+//	node, err := json.Unmarshal([]byte(`{"key": "value"}`))
+//	if err != nil {
+//		ufmt.Println(err)
+//	}
+//	println(node) // {"key": "value"}
+func Unmarshal(data []byte) (*Node, error) {
```

Comment:
Is there any chance we can simplify this method, or break it up into smaller chunks that are easier to read?

Reply:
Well, I tried to reduce the size of this function as much as possible, but it's quite challenging. I think I need to reflect on my past self.